### PR TITLE
AP1-1767 Remove html attribute pattern to prevent html form validation

### PR DESCRIPTION
**Pull request by Obsiye**
## AP1-1767 Remove html attribute pattern to prevent html form validation

[Link to story](https://dsdmoj.atlassian.net/browse/AP-1767)

Remove html attribute pattern to prevent html form validation because we use our form object to validate not the html as it was not showing gds style errors.

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by colinbruce — COMMENTED** [review]
@mpw5 could you take a look at this?  I'm pretty sure we added this after a DAC recommendation https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/1824

**Comment by mpw5** [comment]
@colinbruce @Obsiye We did indeedv add this based on a DAC recommendation. The ticket is [here](https://dsdmoj.atlassian.net/browse/AP-1629) and the original PR is [here](https://github.com/ministryofjustice/laa-apply-for-legal-aid/pull/1824).

The recommendation from DAC was that we add `inputmode="numeric" pattern="[0-9]*"` as the previous version did not automatically display a numeric keypad on mobile devices, which meant that mobile users had to manually switch keyboards to enter dates. I _think_ the `inputmode="numeric"` part of that change is the key for adding that behaviour - `pattern="[0-9]*` is validation and if we're already validating the input in the form then it's probably ok to remove it.

Basically I'd suggest testing this on a mobile device and if the numeric keyboard is displayed on date fields _and_ errors on those fields all render correctly then it should be good to go.

**Review by colinbruce — APPROVED** [review]
Have tested on mobile and this works as expected
Date fields display numeric keyboard and error messages are displayed
image
